### PR TITLE
GTC-2171 - Test Raster Tile Cache Assets Function

### PR DESCRIPTION
**Pull request by gtempus**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [x] Other (please describe): 
  - [x] Adding tests to the `raster_tile_cache_asset` function

## What is the current behavior?

Issue Number: [GTC-2171](https://gfw.atlassian.net/browse/GTC-2171)

## What is the new behavior?

Test coverage that asserts the various collaborator interfaces that are used within `raster_tile_cache_asset`.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by gtempus** [comment]
Renamed the branch so that it could be deployed. The old draft PR is #357

**Comment by gtempus** [comment]
Moved to PR #361

**Comment by gtempus** [comment]
Doh! Reopening this..... 🙄

File: tests_v2/unit/app/tasks/raster_tile_cache_assets/collaborators/test_task_execution_collaboration.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+        args, _ = execute_mock.call_args_list[-1]
+        result = [job.job_name for job in args[0]]
+        assert result == [
+            "reprojection_job",  # zoom level 1
+            "symbology_job",  # zoom level 1
+            "reprojection_job",  # zoom level 0
+            "symbology_job",  # zoom level 0
+            "tile_cache_job",  # zoom level 0 - b/c zoom level is <= max static zoom
+        ]
```
Comment (dmannarino) [inline tests_v2/unit/app/tasks/raster_tile_cache_assets/collaborators/test_task_execution_collaboration.py:86]: Thanks for labeling these!

Comment (gtempus) [reply tests_v2/unit/app/tasks/raster_tile_cache_assets/collaborators/test_task_execution_collaboration.py:86]: Yeah! I'd love to make those jobs more self-descriptive in the future. I may do that in later refactors.

**Review by dmannarino — APPROVED** [review]
Awesome, thanks!

**Pull request merged**